index.html

```diff
@@ -5099,6 +5125,31 @@ <h2 id="defining-a-valid-audioworkletprcoessor">
                 <code>"name"</code>.
               </li>
             </ol>
+            <p>
+              The return value of this method controls the lifetime of the
+              <a>AudioWorkletProcessor</a>'s associated <a>AudioWorkletNode</a>
+              by determining the existence of the node's <a href=
+              "#dfn-active-reference">active reference</a>. At the conclusion
+              of each call to the <code>process()</code> method, if the result
+              of applying <a href=
+              "https://tc39.github.io/ecma262/#sec-toboolean"><code>ToBoolean</code></a>
+              (described in [[!ECMASCRIPT]]) to the return value is equal to
+              <code>true</code>, then a single <a>active reference</a> is
+              retained to the node, affecting the node's <a href=
+              "#lifetime-AudioNode">lifetime</a>. Otherwise, any active
+              reference is removed and the node's lifetime depends only on
+              other references as they may exist.
+            </p>
+            <div class="note">
+              While returning <code>true</code> ensures that
+              <code>process()</code> will be called during subsequent rendering
+              quanta, returning <code>false</code> does not prevent future
+              calls to <code>process()</code>. Other references to the
+              corresponding <a>AudioWorkletNode</a> may exist and the node may
+              thus continue to function. Furthermore, these future calls may
+              return a value of <code>true</code>, reinstating the active
+              reference.
```

Comment:
Now I see why we're confused about this issue, but then we have to think about wording 'reinstating'. I don't think it is reinstating the node. It was not pulled off from the graph because we didn't care about its `false` return value before two previously mentioned conditions were not met.

Reinstating means you're keeping the node's JS reference somewhere. Right? Without that reinstating should not be possible at all.

> However you're making an additional assertion here, which is that the UA must check immediately after a process() invocation whether all the references have gone away.

UA always gets the return value of process() method immediately. It's not like we can defer this operation somehow. This has nothing to do with the timing of the actual GC.

Let me clarify one more time:

- JS Reference to an AWN is dropped.
- There is no active input to an AWP.
- AWP::process() is `false` at the render quantum Q.

Then,
- This node is disconnected before the render quantum Q+1. Thus `process()` method will never get called again.
- This pair of AWN/AWP instance is marked for GC.